magda-web-client/src/reducers/datasetSearchReducer.js

```diff
@@ -92,7 +92,7 @@ const datasetSearchReducer = (
                 data && data.dataSets && data.dataSets;
             let hitCount: number = data && data.hitCount && data.hitCount;
             let temporalRange: Array<Object> =
-                data && data.temporal
+                data && (data.temporal.start && data.temporal.end)
```

Comment:
Although itl is promised that data.tempora must has a value, change 
```
data && (data.temporal.start && data.temporal.end)
```
to 
```
(data && data.temporal && data.temporal.start && data.temporal.end)
```
would be more robust and cover any future changes --- just in case

Reply:
yes you are right! thanks!